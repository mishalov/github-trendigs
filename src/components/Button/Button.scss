@import "../../styles/variables.scss";
.button {
  color: #fff;
  background-color: $main-ui-color;
  border-radius: 3px;
  padding: 8px 12px;
  border-style: none;

  &--disabled {
    background-color: $col-secondary-element;
    color: $col-primary-font;
  }
}
