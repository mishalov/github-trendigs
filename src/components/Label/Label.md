Компонент не используется сам по себе, нужен для описания полей ввода

```jsx
<Label>Пример использования</Label>
```
