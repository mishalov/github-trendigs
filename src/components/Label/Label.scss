@import '../../styles/variables.scss';

.input-label {
  display: block;
  font-size: 14px;
  line-height: 20px;
  height: 20px;
  font-family: GraphikRBCLC;
  font-weight: bold;
  margin-bottom: 8px;
}
