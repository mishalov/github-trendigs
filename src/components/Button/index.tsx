import React from "react";
import "./Button.scss";

interface IButtonProps {
  onClick: () => void;
}

const Button: React.FC<IButtonProps> = props => {
  const { onClick, children } = props;
  return (
    <button className="button" onClick={onClick}>
      {children}
    </button>
  );
};

export default Button;
