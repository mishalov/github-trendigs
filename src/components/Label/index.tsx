import React from "react";
import cn from "classnames";
import "./Label.scss";

export default (props: React.HTMLAttributes<HTMLLabelElement>) => (
  <label {...props} className={cn("input-label", props.className)}>
    {props.children}
  </label>
);
