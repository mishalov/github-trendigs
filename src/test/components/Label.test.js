import React from "react";
import { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import Enzyme from "enzyme";
import Button from "../../components/Button";
import Label from "../../components/Label";

Enzyme.configure({ adapter: new Adapter() });
const wrappers = [];

beforeEach(() => {
  wrappers.push(shallow(<Label>test</Label>));
  wrappers.push(
    shallow(
      <Label className="test-class">
        <Button>test</Button>
      </Label>
    )
  );
});

describe("<Label /> rendering", () => {
  it("renders correctly", () => {
    expect(wrappers[0]).toMatchSnapshot();
  });
  it("attributes test", () => {
    expect(wrappers[1]).toMatchSnapshot();
  });
});
