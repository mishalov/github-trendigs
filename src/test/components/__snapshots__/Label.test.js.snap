// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Label /> rendering attributes test 1`] = `
<label
  className="input-label test-class"
>
  <Button>
    test
  </Button>
</label>
`;

exports[`<Label /> rendering renders correctly 1`] = `
<label
  className="input-label"
>
  test
</label>
`;
